subworkflows/nf-neuro/io_bids/main.nf

```diff
@@ -0,0 +1,88 @@
+include { IO_READBIDS   } from '../../../modules/nf-neuro/io/readbids/main.nf'
+
+workflow IO_BIDS {
+
+    take:
+    bids_folder         // channel: [ path(bids_folder) ]
+    fs_folder           // channel: [ path(fs_folder) ]
+    bidsignore         // channel: [ path(bids_ignore) ]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // ** Fetching the BIDS data as a json file ** //
+    IO_READBIDS (
+        bids_folder,
+        fs_folder,
+        bidsignore
+    )
+    ch_versions = ch_versions.mix(IO_READBIDS.out.versions)
+
+    // ** Converting the json file into channels. ** //
+    ch_files = IO_READBIDS.out.bidsstructure
+        .flatMap{ layout ->
+            def json = new groovy.json.JsonSlurper().parseText(layout.getText())
+            json.collect { item ->
```

Comment:
Just to clarify this statement, add a `return` before. I know collect does it automatically, but I think it could mislead the reader.